src/Microsoft.Health.Dicom.Core/Features/Store/ValidationResultBuilderFactory.cs

```diff
@@ -0,0 +1,24 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+
+internal class ValidationResultBuilderFactory : IValidationResultBuilderFactory
```

Comment:
What are we using this factory for? I think we can just create the validation result.

Reply:
ValidationResult/Builder has to be transient. And I was trying to avoid "new..." in the code, to make our life easier with the unit test.

Reply:
Removed DI. Kept the builder, to abstract details on how the errors/warnings are collected. Builder.Build returns an immutable DTO instance.